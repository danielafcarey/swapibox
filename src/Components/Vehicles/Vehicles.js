import React from 'react';
import PropTypes from 'prop-types';

import Card from '../Card/Card.js';

const Vehicles = ({ cardData, toggleFavorite }) => {
  const vehiclesCards = cardData.map(card => {
    const { 
      id, 
      favorite, 
      name, 
      vehicleClass, 
      model, 
      numberOfPassengers 
    } = card;

    return (
      <Card
        id={ id }
        title={ name }
        item1={ `Class: ${vehicleClass}` }
        item2={ `Model: ${model}` }
        item3={ `Total Passengers: ${numberOfPassengers}` }
        favorite={ favorite }
        section='Vehicles'
        toggleFavorite={ toggleFavorite }
        key={ id }
      />
    );
  });

  return (
    <div className="card-container Vehicles">
      { vehiclesCards }
    </div>
  );
};

Vehicles.propTypes = {
  cardData: PropTypes.arrayOf(PropTypes.object).isRequired,
  toggleFavorite: PropTypes.func.isRequired
};

export default Vehicles;
