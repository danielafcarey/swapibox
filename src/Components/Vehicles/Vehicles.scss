.Vehicles {
  .Card {
    background: rgba(43, 43, 43, 0.65) url('../../images/vehicles.png') no-repeat center;
  }
}
