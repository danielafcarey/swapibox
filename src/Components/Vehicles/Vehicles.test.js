import React from 'react';
import { shallow } from 'enzyme';
import Vehicles from './Vehicles';

describe('Vehicles', () => {

  it('matches the snapshot with one card', () => {
    const mockCardData = [{
      id: 1,
      favorite: false,
      name: 'Daniela',
      vehicleClass: 'car',
      model: 'cool',
      numberOfPassengers: 1
    }];
    const wrapper = shallow(
      <Vehicles
        cardData={ mockCardData }
        toggleFavorite={ jest.fn() }
      /> 
    );

    expect(wrapper).toMatchSnapshot();
  })

  it('matches the snapshot with cards', () => {
    const mockCardData = [{
      id: 1,
      favorite: false,
      name: 'Daniela',
      vehicleClass: 'car',
      model: 'cool',
      numberOfPassengers: 1
    },
      {
        id: 1,
        favorite: false,
        name: 'Daniela',
        vehicleClass: 'car',
        model: 'cool',
        numberOfPassengers: 1
      }
    ];
    const wrapper = shallow(
      <Vehicles
        cardData={ mockCardData }
        toggleFavorite={ jest.fn() }
      /> 
    );

    expect(wrapper).toMatchSnapshot();
  })

});

