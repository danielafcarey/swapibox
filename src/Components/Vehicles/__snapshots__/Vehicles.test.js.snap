// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles matches the snapshot with cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    cardData={
        Array [
            Object {
              "favorite": false,
              "id": 1,
              "model": "cool",
              "name": "Daniela",
              "numberOfPassengers": 1,
              "vehicleClass": "car",
            },
            Object {
              "favorite": false,
              "id": 1,
              "model": "cool",
              "name": "Daniela",
              "numberOfPassengers": 1,
              "vehicleClass": "car",
            },
          ]
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorite={false}
          id={1}
          item1="Class: car"
          item2="Model: cool"
          item3="Total Passengers: 1"
          title="Daniela"
          toggleFavorite={[Function]}
/>,
        <Card
          favorite={false}
          id={1}
          item1="Class: car"
          item2="Model: cool"
          item3="Total Passengers: 1"
          title="Daniela"
          toggleFavorite={[Function]}
/>,
      ],
      "className": "card-container Vehicles",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "favorite": false,
          "id": 1,
          "item1": "Class: car",
          "item2": "Model: cool",
          "item3": "Total Passengers: 1",
          "title": "Daniela",
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "favorite": false,
          "id": 1,
          "item1": "Class: car",
          "item2": "Model: cool",
          "item3": "Total Passengers: 1",
          "title": "Daniela",
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorite={false}
            id={1}
            item1="Class: car"
            item2="Model: cool"
            item3="Total Passengers: 1"
            title="Daniela"
            toggleFavorite={[Function]}
/>,
          <Card
            favorite={false}
            id={1}
            item1="Class: car"
            item2="Model: cool"
            item3="Total Passengers: 1"
            title="Daniela"
            toggleFavorite={[Function]}
/>,
        ],
        "className": "card-container Vehicles",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "favorite": false,
            "id": 1,
            "item1": "Class: car",
            "item2": "Model: cool",
            "item3": "Total Passengers: 1",
            "title": "Daniela",
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "favorite": false,
            "id": 1,
            "item1": "Class: car",
            "item2": "Model: cool",
            "item3": "Total Passengers: 1",
            "title": "Daniela",
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Vehicles matches the snapshot with one card 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    cardData={
        Array [
            Object {
              "favorite": false,
              "id": 1,
              "model": "cool",
              "name": "Daniela",
              "numberOfPassengers": 1,
              "vehicleClass": "car",
            },
          ]
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorite={false}
          id={1}
          item1="Class: car"
          item2="Model: cool"
          item3="Total Passengers: 1"
          title="Daniela"
          toggleFavorite={[Function]}
/>,
      ],
      "className": "card-container Vehicles",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "favorite": false,
          "id": 1,
          "item1": "Class: car",
          "item2": "Model: cool",
          "item3": "Total Passengers: 1",
          "title": "Daniela",
          "toggleFavorite": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorite={false}
            id={1}
            item1="Class: car"
            item2="Model: cool"
            item3="Total Passengers: 1"
            title="Daniela"
            toggleFavorite={[Function]}
/>,
        ],
        "className": "card-container Vehicles",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "favorite": false,
            "id": 1,
            "item1": "Class: car",
            "item2": "Model: cool",
            "item3": "Total Passengers: 1",
            "title": "Daniela",
            "toggleFavorite": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
